### PR TITLE
Sprint 3

### PR DESCRIPTION
**Pull request by iHamsin**

### CODE REVIEW

**Comment by arttet** [comment]
Please add the CodeCov Coverage Report

File: internal/repositories/interfaces.go [hunk 1]
```diff
@@ -13,16 +15,41 @@ type Repository interface {
...
+
+		db, postgresOpenError := pgx.Connect(context.Background(), cfg.Repository.DatabaseDSN)
+		if postgresOpenError != nil {
+			outError = postgresOpenError
+			repository = nil
+		} else {
+			repository = NewLinksRepoPGSQL(db)
+
+			createTableSQL := `
+				CREATE TABLE IF NOT EXISTS  "public"."links" (
```
Comment (arttet) [inline internal/repositories/interfaces.go:38]: Please use migrations.

File: internal/repositories/linkRepoInPGSQL.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+// 	OriginalURL string `json:"original_url"`
+// }
+
+// New -.
+func NewLinksRepoPGSQL(db *pgx.Conn) *linkRepoInPGSQL {
+	return &linkRepoInPGSQL{db}
+}
+
+// Insert -.
+func (r *linkRepoInPGSQL) Insert(originalURL string) (string, error) {
```
Comment (arttet) [inline internal/repositories/linkRepoInPGSQL.go:32]: You should provide a context here.

Comment (arttet) [inline internal/repositories/linkRepoInPGSQL.go:32]:
```suggestion
func (r *repo) InsertUrl(originalURL string) (string, error) {
```

**Review by arttet — CHANGES_REQUESTED** [review]
Please add more tests

File: internal/repositories/interfaces.go [hunk 3]
```diff
@@ -1,28 +1,65 @@
...
-	GetByCode(string) (string, error)
-	Insert(string) (string, error)
+	GetLinkByCode(context.Context, string) (string, error)
+	InsertLink(context.Context, string) (string, error)
+	Check() error
 	Close()
+	BatchInsertLink(context.Context, []string) ([]string, error)
 }
 
+//go:embed migrations/*.sql
```
Comment (arttet) [inline internal/repositories/interfaces.go:28]: Please move it to the `migrations` dir.

File: internal/repositories/migrations/00001_init.up.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+-- );
+
+CREATE TABLE "public"."links" (
+  "original_link" text COLLATE "pg_catalog"."default" NOT NULL,
+  "short_link" text COLLATE "pg_catalog"."default" NOT NULL,
+  UNIQUE(original_link)
+)
+;
+
+CREATE INDEX "original_link" ON "public"."links" USING btree (
```
Comment (arttet) [inline internal/repositories/migrations/00001_init.up.sql:14]: Why do you use `btree` indexes?

**Review by arttet — APPROVED** [review]

File: internal/repositories/migrations/00001_init.up.sql (hunk 4 above)
Comment (iHamsin) [reply internal/repositories/migrations/00001_init.up.sql:14]: Unusable DB index deleted in new migration

**Pull request merged**